### PR TITLE
Support vSphere VM tagging

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Description of changes:*
Add tags option to vSphereMachineConfig so users can assign vSphere tags to their VM's during the EKS-A deployment.

This PR is just a cleanup of the original PR here: https://github.com/aws/eks-anywhere/pull/3786

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by gwesterfieldjr** [comment]
Looks good to me! Haha. I love you all and miss you all thanks for finally getting this in!

**Comment by abhinavmpandey08** [comment]
> Looks good to me! Haha. I love you all and miss you all thanks for finally getting this in!

I removed some of those pesky linter fixes. I hope you don't mind 😂
Missing you as well! You gotta pull up to Seattle some time!

File: pkg/providers/vsphere/validator.go [hunk 1]
```diff
@@ -89,7 +89,31 @@ func (v *Validator) ValidateVCenterConfig(ctx context.Context, datacenterConfig
...
 }
 
-// TODO: dry out machine configs validations.
+func (v *Validator) validateMachineConfigTagsExist(ctx context.Context, machineConfigs []*anywherev1.VSphereMachineConfig) error {
+	tags, err := v.govc.ListTags(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if tags exists in vSphere: %v", err)
+	}
+
+	tagIds := make([]string, 0, len(tags))
```
Comment (jacobweinstock) [inline pkg/providers/vsphere/validator.go:98]: might think about standardizing here. `tagIds` -> `tagIDS`

ref: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/providers/vsphere/validator.go [hunk 2]
```diff
@@ -89,7 +89,31 @@ func (v *Validator) ValidateVCenterConfig(ctx context.Context, datacenterConfig
...
+	idLookup := types.SliceToLookup(tagIds)
+	for _, machineConfig := range machineConfigs {
+		for _, tagID := range machineConfig.Spec.TagIDs {
+			if !idLookup.IsPresent(tagID) {
+				return fmt.Errorf("tag (%s) does not exist in vSphere. please provide a valid tag id in the urn format (example: urn:vmomi:InventoryServiceTag:8e0ce079-0677-48d6-8865-19ada4e6dabd:GLOBAL)", tagID)
+			}
+		}
+	}
+
+	logger.MarkPass("Machine config tags validated")
```
Comment (jacobweinstock) [inline pkg/providers/vsphere/validator.go:112]: should there be a space after this logger line? looks like most of the other funcs have a space before the final return.

File: pkg/providers/vsphere/validator.go [hunk 3]
```diff
@@ -89,7 +89,31 @@ func (v *Validator) ValidateVCenterConfig(ctx context.Context, datacenterConfig
...
+	return nil
+}
+
+// ValidateClusterMachineConfigs TODO: dry out machine configs validations.
```
Comment (jacobweinstock) [inline pkg/providers/vsphere/validator.go:116]: what is "dry out"? "dont repeat yourself"?

**Pull request merged**